CONTRIBUTING.md

```diff
@@ -1,103 +1,117 @@
-# Wandb library
-
-## Get the code / library
-
-Checkout from github:
+# Contributing to `wandb`
+
+We at Weights & Biases ❤️ open source and welcome contributions from the community!
+This guide discusses the development workflow and the internals of the `wandb` client library.
+
+## Development workflow
+
+1. If you are a first-timer, please go to `https://github.com/wandb/client` 
+and click the "Fork" button in the top-right corner of the page. 
+This will create your personal copy of the repository that you will use for development. 
+   - Set up [SSH authentication with GitHub](https://docs.github.com/en/authentication/connecting-to-github-with-ssh).
+   - Clone the forked project to your machine and add the upstream repository 
+     that will point to the main `wandb` project:
+    ```shell
+    git clone https://github.com/<your-username>/client.git
+    cd client
+    git remote add upstream https://github.com/wandb/client.git   
+    ```
+
+2. Develop you contribution.
+   - Make sure your fork is in sync with the main repository:
+    ```shell
+    git checkout master
+    git pull upstream master   
+    ```
+   - Create a `git` branch where you will develop your contribution. 
+     Use a sensible name for the branch, for example:
+    ```shell
+    git checkout -b new-awesome-feature
+    ```
+   - Hack! As you make progress, commit your changes locally, e.g.:
+    ```shell
+    git add changed-file.py tests/test-changed-file.py
+    git commit -m "Added integration with a new library"
+    ```
+   - Test and lint your code! Please see below for a detailed discussion.
```

Comment:
maybe add hyperlink to the relevant section?

Reply:
Added.